src/shogun/classifier/mkl/MKLClassification.cpp

```diff
@@ -42,11 +42,12 @@ CMKLClassification* CMKLClassification::obtain_from_generic(CMachine* machine)
 {
 	if (machine == NULL)
 		return NULL;
-
-	if (machine->get_classifier_type() != CT_MKLCLASSIFICATION)
-		SG_SERROR("Provided machine is not of type CMKLClassification!")
-
+		
 	CMKLClassification* casted = dynamic_cast<CMKLClassification*>(machine);
```

Comment:
just remove the obtain_from_generic method completely. Wherever it was called, replace the call with `->as<CMKLClassification>()`